AutomaticQC/imosTimeSeriesSpikeQC.m

```diff
@@ -47,18 +47,11 @@
 
 if (is_profile || time_missing), return, end
 
-has_burst_duration = false;
-has_burst_interval = false;
-try
-    has_burst_duration = sample_data.meta.instrument_burst_duration;
-    has_burst_interval = sample_data.meta.instrument_burst_interval;
-catch
-    try
-        has_burst_duration = sample_data.instrument_burst_duration;
-        has_burst_interval = sample_data.meta.instrument_burst_interval;
-    end
-end
-has_burst = has_burst_duration || has_burst_interval;
+has_burst_duration = (isfield(sample_data.meta, 'instrument_burst_duration') && ~isempty(sample_data.meta.instrument_burst_duration) && ~isnan(sample_data.meta.instrument_burst_duration)) || (isfield(sample_data, 'instrument_burst_duration') && ~isempty(sample_data.instrument_burst_duration) && ~isnan(sample_data.instrument_burst_duration));
```

Comment:
isn't ~isnan(sample_data.meta.instrument_burst_duration)  always the case?

Reply:
No strict guarantees, since there is no schema validation for that structure. All sample_data fields can be modified by the user in the UI, or be anything in a netcdf for example.